File: C++/two_sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> numToInd;

        for (int i = 0; i < nums.size(); i++){
            if (numToInd.contains(target - nums[i])){
                vector<int> res = {numToInd[target - nums[i]], i};
                return res;
            }
            numToInd[nums[i]] = i;
        }

        return {};
    }
};

File: python/two_sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        numToInd = {}

        for i, n in enumerate(nums):
            if target - n in numToInd:
                return [i, numToInd[target - n]]
            numToInd[n] = i
